assets/src/edit-story/components/library/panes/media/media3p/media3pCategories.js

```diff
@@ -33,19 +34,32 @@ const CategorySection = styled.div`
   flex-wrap: wrap;
 `;
 
-// TODO(#2362) Wire up pill list to state and remove these fake pills.
-const fakePillList = ['COVID-19', 'Nature', 'Wallpapers', 'People'];
-const selectedPill = 'Nature';
-
-const Media3pCategories = () => (
+const Media3pCategories = ({
+  categories,
+  selectedCategory,
+  selectCategory,
+  deselectCategory,
+}) => (
   <CategorySection>
-    {
-      // TODO(#2362) Wire up pill list to state and remove these fake pills.
-      fakePillList.map((e, i) => (
-        <CategoryPill isSelected={e == selectedPill} key={i} title={e} />
-      ))
-    }
+    {categories.map((e) => {
+      const selected = e.name === selectedCategory;
+      return (
+        <CategoryPill
+          isSelected={selected}
+          key={e.name}
+          title={e.displayName}
+          onClick={selected ? deselectCategory : () => selectCategory(e.name)}
```

Comment:
Please add a unit test to make sure that when clicked, it is triggering the either "deselectCategory" or "selectCategory" based on "selected" state.

Reply:
Done.